posts/2024/sdk_12_4_0_ga.md

```diff
@@ -0,0 +1,42 @@
+---
+title: 'Titanium SDK 12.4.0.GA released'
+date: '2024-07-17'
+category: 'Release'
+author: 'Hans Knöchel'
+teaser: 'Android API 34 Support'
+image: https://tidev.io/images/titanium-general.png
+---
+
+![Titanium SDK 12.4.0.GA](/images/titanium-general.png)
+
+The next version of the Titanium SDK is now is available: <b>12.4.0.GA</b>. Titanium SDK 12.4.0 is a minor release of the SDK, adding new features and platform updates.
+
+This update is focussed on supporting the latest Android API level 34, which will be required by Google for all new apps and app updates starting in August 2024.
+To retain backwards compatibility in this minor release, the feature will be opt-in until the release of Titanium SDK 13.0.0, where it will become the new default value.
+
+To opt-in for API level 34, add the following to your tiapp.xml:
+```xml
+ <android>
+    <manifest>
+        <uses-sdk android:targetSdkVersion="34" />
+    </manifest>
+</android>
+```
+
```

Comment:
## Features

Besides the Android API level 34 update we have some other highlights in this release:

* Android/iOS: Swipe actions support for Ti.UI.TableView ([demo video](https://github.com/tidev/titanium-sdk/pull/14065))
* Android: new moveToBackground() method to move the whole app to the background
* Android: option to hide scrollbars in a WebView
* Android: missing Event.remove() method was added
* Android: parity for OptionBar color properties<br/><img src="https://media.dev.to/cdn-cgi/image/width=800%2Cheight=%2Cfit=scale-down%2Cgravity=auto%2Cformat=auto/https%3A%2F%2Fdev-to-uploads.s3.amazonaws.com%2Fuploads%2Farticles%2Fpsn9xuv5avfhifk8d0qj.png"/>
* Android: change track colors in a Ti.UI.Switch<br/><img src="https://media.dev.to/cdn-cgi/image/width=800%2Cheight=%2Cfit=scale-down%2Cgravity=auto%2Cformat=auto/https%3A%2F%2Fdev-to-uploads.s3.amazonaws.com%2Fuploads%2Farticles%2F3onxi1vx4ak561syzfpw.png"/>
* Android: text alignment for date pickers<br/><img src="https://media.dev.to/cdn-cgi/image/width=800%2Cheight=%2Cfit=scale-down%2Cgravity=auto%2Cformat=auto/https%3A%2F%2Fdev-to-uploads.s3.amazonaws.com%2Fuploads%2Farticles%2Fk31br3wbgfg815ppxdb1.png"/>
* Android: change video playback speed
* Android: defaultLang option in tiapp.xml (in case you run an app that doesn't have EN as the first language)
* iOS: iOS 17 symbol effects ([demo video](https://github.com/tidev/titanium-sdk/pull/13982))
* iOS: change the backgroundColor of a RefreshControl<br/><img src="https://media.dev.to/cdn-cgi/image/width=800%2Cheight=%2Cfit=scale-down%2Cgravity=auto%2Cformat=auto/https%3A%2F%2Fdev-to-uploads.s3.amazonaws.com%2Fuploads%2Farticles%2F3t7c9m8a9yg4c8q3v1p6.png"/>
* iOS: use overrideUserInterfaceStyle for a Picker